src/core/memory.cc

```diff
@@ -1890,7 +1890,9 @@ void* realloc(void* ptr, size_t size) {
     if (try_trigger_error_injector()) {
         return nullptr;
     }
-    if (ptr != nullptr && !is_seastar_memory(ptr)) {
```

Comment:
"If ptr is a null pointer, the behavior is the same as calling std::malloc(new_size)."

sounds like a good comment to place here, so future readers can understand why we're bypassing the realloc impl when ptr == null.

Reply:
Fixed in v2